Move vector embeddings to HANA section (#768)
cap-js/docs: guides/databases-hana.md

The HANA Service provides dedicated support for native SAP HANA features as follows.

### Vector Embeddings { #vector-embeddings }

Vector embeddings are numerical representations that capture features and inherent semantics of unstructured data - such as text, images, or audio. They facilitate tasks like similarity search, recommendations, and Retrieval Augmented Generation (RAG). This improves the results of generative AI (GenAI) by augmenting prompts with relevant data retrieved from a vector datastore such as the [SAP HANA Cloud Vector Engine](https://community.sap.com/t5/technology-blogs-by-sap/sap-hana-cloud-s-vector-engine-announcement/ba-p/13577010).

Typically vector embeddings are computed using models tailored to a specific use case, like large language models (LLMs) for text, or convolutional neural networks (CNNs) for images. The dimensionality of the vector embedding space depends on the chosen model. Unified LLM consumption across different vendors and open source models is provided via the [SAP Generative AI Hub](https://community.sap.com/t5/technology-blogs-by-sap/how-sap-s-generative-ai-hub-facilitates-embedded-trustworthy-and-reliable/ba-p/13596153).

> [Learn more about Vector Embeddings in CAP Java](../java/cds-data#vector-embeddings) {.learn-more}

Reviewer: 👍 I'll create a similar link for Node.js

Typically vector embeddings are computed using models tailored to a specific use case, like large language models (LLMs) for text, or convolutional neural networks (CNNs) for images. The dimensionality of the vector embedding space depends on the chosen model. Unified LLM consumption across different vendors and open source models is provided via the [SAP Generative AI Hub](https://community.sap.com/t5/technology-blogs-by-sap/how-sap-s-generative-ai-hub-facilitates-embedded-trustworthy-and-reliable/ba-p/13596153).

In CAP, vector embeddings are stored in elements of type [cds.Vector](../cds/types.md#built-in-types):

```cds
entity Books : cuid { // [!code focus]
  title         : String(111);
  embedding     : Vector(1536); // vector space w/ 1536 dimensions // [!code focus]
} // [!code focus]
```

> At runtime, you can use it leveraging the Query API:

Author: or
```suggestion
At runtime, you can use them in queries:
```

::: code-group
```js [Node.js]
let embedding; // vector embedding as string '[0.3,0.7,0.1,...]';

let relatedBooks = await SELECT.from('Books')
  .where`cosine_similarity(embedding, to_real_vector(${embedding})) > 0.9`
```

```java [Java]

> //  Compute vector embedding, e.g. via LangChain4j
> float[] embedding = llm.embed(text).content().vector();
> 
> CqnSelect query = Select.from(BOOKS).where(b -> // [!code focus]
>   CQL.cosineSimilarity(b.embedding(), CQL.vector(embedding)).gt(0.9)) // [!code focus]
> 
> Result relatedBooks = service.run(query); // [!code focus]

Author: ```suggestion
// Vector embedding of text, e.g. from SAP GenAI Hub or via LangChain4j
float[] embedding = llm.embed(text).content().vector();

CqnSelect query = Select.from(BOOKS).where(b -> 
  CQL.cosineSimilarity(b.embedding(), CQL.vector(embedding)).gt(0.9))

Result relatedBooks = service.run(query);
```
